### PR TITLE
[RLlib] Fix RolloutWorker filter syncing

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

When creating filter connectors, these are not properly added to the RolloutWorker's filters.
That is because we scanned for them by class name, rather than with issubclass.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/connectors/util.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def maybe_get_filters_for_syncing(rollout_worker, policy_id):
     ]
     # There can only be one filter at a time
     if filter_connectors:
-        assert len(SyncedFilterAgentConnector) == 1, (
```
Comment (ArturNiederfahrenhorst) [inline rllib/connectors/util.py:147]: This line was never executed

File: rllib/connectors/connector.py [hunk 2]
```diff
@@ -445,7 +445,11 @@ def __getitem__(self, key: Union[str, int, type]):
             elif isinstance(key, int):
                 return [self.connectors[key]]
             elif isinstance(key, type):
-                key = key.__name__
```
Comment (ArturNiederfahrenhorst) [inline rllib/connectors/connector.py:448]: This makes it so that we can fetch all SynchedFilters.

**Review by gjoliver — APPROVED** [review]
Makes so much sense.
Thanks.

**Comment by gjoliver** [comment]
Test failures look unrelated

**Pull request merged**